### PR TITLE
RPC Wallet Api

### PR DESCRIPTION
**Pull request by m-aboelenein**
Examples for using: 

- Providers Methods
- RPC Methods (request/listen)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -11,7 +11,7 @@
     "react": "^18.2.0",
     "react-dom": "^18.2.0",
     "react-scripts": "5.0.1",
-    "sats-connect": "1.4.0",
+    "sats-connect": "file:../sats-connect",
```
Comment (aryzing) [inline package.json]: Should be updated to non-local version?

Comment (m-aboelenein) [reply package.json]: my bad must have forgot to replace it

**Review by DuskaT021 — DISMISSED** [review]

File: src/App.tsx [hunk 2]
```diff
@@ -114,11 +145,50 @@ function App() {
...
+          (address) => address.purpose === AddressPurpose.Stacks
+        );
+        setStacksAddress(stacksAddressItem?.address);
+        setStacksPublicKey(stacksAddressItem?.publicKey);
       },
       onCancel: () => alert("Request canceled"),
     });
   };
 
+  const onConnectRPCClick = async () => {
```
Comment (victorkirov) [inline src/App.tsx]: Looks like this won't set the stacks address, which means that isReady above won't be set to true in this scenario. I think once we have the btc addresses, we should do a second RPC call for the stacks address maybe?

File: src/App.css [hunk 3]
```diff
@@ -8,3 +8,23 @@ button {
...
+.providerImg {
+  width: 30px;
+  height: 30px;
+  margin-right: 10px;
+}
+
+.providerName {
+  color: blue;
+  font-size: 16px;
+}
```
Comment (victorkirov) [inline src/App.css]: Please add EOL at the end of files 👀  You can make VSCode insert it automatically on save.

File: src/components/signMessage.tsx [hunk 4]
```diff
@@ -25,10 +30,33 @@ const SignMessage = ({ network, address, capabilities }: Props) => {
...
+        const error = response;
+        console.log(error);
+        if (error.error.code === RpcErrorCode.USER_REJECTION) {
+          alert("Canceled");
+        } else {
+          alert(error.error.message);
+        }
+      }
+    } catch (err) {
+      alert("Something Wen Wrong");
```
Comment (victorkirov) [inline src/components/signMessage.tsx]:
```suggestion
      alert("Something Went Wrong");
```

File: src/components/signTransaction.tsx [hunk 5]
```diff
@@ -95,6 +100,67 @@ const SignTransaction = ({
     );
   }
 
+  const onSignTransactionRPC = async () => {
```
Comment (victorkirov) [inline src/components/signTransaction.tsx:81]: Lot's of duplicate code with the other sign transaction function. Could we combine them with an arg in the function specifying whether to use RPC or not?

File: src/components/stacks/callContract.tsx [hunk 6]
```diff
@@ -0,0 +1,112 @@
...
+
+  const handleWebBtcCallContractClick = async () => {
+    try {
+      const response = await request("stx_callContract", {
+        contract: `${contractAddress}.${contractName}`,
+        functionName,
+        arguments: JSON.parse(functionArgs),
+      });
+      if (response.status === "success") {
+        console.log(response.result);
```
Comment (victorkirov) [inline src/components/stacks/callContract.tsx:54]: `alert("Success! Check console for result.");`

File: src/App.tsx (hunk 2 above)
Comment (aryzing) [reply src/App.tsx]: Removed the callback, since it's not currently being used.

File: src/components/stacks/signTransaction.tsx [hunk 7]
```diff
@@ -0,0 +1,132 @@
...
+      functionName: "set-stx-buffer",
+      functionArgs: [uintCV(1)],
+      publicKey: props.publicKey,
+    });
+    try {
+      const response = await request("stx_signTransaction", {
+        transaction: uint8ArrayToHex(transaction.serialize()),
+      });
+      if (response.status === "success") {
+        console.log(response.result.transaction);
```
Comment (victorkirov) [inline src/components/stacks/signTransaction.tsx:34]: Maybe a `alert("Success! Check console for result."); ` here

File: src/App.tsx [hunk 8]
```diff
@@ -75,13 +90,24 @@ function App() {
...
     setPaymentPublicKey(undefined);
     setOrdinalsAddress(undefined);
     setOrdinalsPublicKey(undefined);
+    setStacksAddress(undefined);
+  };
+
+  const handleGetInfo = async () => {
+    try {
+      const response = await request("getInfo", null);
+      console.log(response);
```
Comment (victorkirov) [inline src/App.tsx]: Alert here as well 🙏

Comment (victorkirov) [reply src/App.tsx]: Or display it in the component maybe

File: src/components/signTransaction.tsx (hunk 5 above)
Comment (aryzing) [reply src/components/signTransaction.tsx:81]: Fixed, moved dupe code to new fn

File: src/components/signTransaction/index.tsx [hunk 9]
```diff
@@ -105,9 +128,19 @@ const SignTransaction = ({
...
 export default SignTransaction;
+function createTransaction(arg0: {
+  network: BitcoinNetworkType;
+  paymentAddress: string;
+  ordinalsAddress: string;
+  paymentPublicKey: string;
+  ordinalsPublicKey: string;
+}): [any, any] | PromiseLike<[any, any]> {
+  throw new Error("Function not implemented.");
+}
```
Comment (victorkirov) [inline src/components/signTransaction/index.tsx]: I think this may have been the template for the util which needs to be imported still 👀

Comment (aryzing) [reply src/components/signTransaction/index.tsx]: Good catch!

**Review by DuskaT021 — APPROVED** [review]

**Pull request merged**